### PR TITLE
feat(KFLUXVNGD-1116): add normalized_uri label to access-log-exporter metrics

### PR DESCRIPTION
**Pull request by amisstea**
Enable per-repository-type metric breakdowns by extracting the first two URI path segments from each request. This controls label cardinality while still distinguishing traffic by repository type (e.g. /repository/go-proxy/.+). Query strings are also stripped to prevent unbounded cardinality from varying query parameters.

Assisted-by: Claude Opus 4.6

### Author's Checklist
- [x] I understand the problem that the PR is trying to address.
- [x] I understand the solution and its role and impact within the wider system.
- [x] I opted for automation and automated tests over documenting multiple steps.

### Reviewer's Guide
- [ ] What is the PR trying to solve?
- [ ] Does the solution make sense?
- [ ] How does this affect the wider system?
- [ ] Is it being tested properly?
- [ ] Does it keep documentation concise and maintainable?

### CODE REVIEW

File: caching/templates/nginx-exporter-configmap.yaml [hunk 1]
```diff
@@ -44,6 +44,16 @@ data:
                 lineIndex: 0
               - name: "status"
                 lineIndex: 1
+              # Keeps the first 2 path segments to group by repository type (e.g. /repository/go-proxy/.+)
+              # and strips query strings to control cardinality. First-match-wins: the second
+              # replacement only applies to URIs with fewer than 3 segments.
+              - name: "normalized_uri"
+                lineIndex: 2
+                replacements:
+                  - regexp: "^(/[^/]+/[^/]+/).+"
```
Comment (yftacherzog) [inline caching/templates/nginx-exporter-configmap.yaml]: Qodo raised this point and it is important: with first-match-wins semantics, `.+` matches `?query` as trailing content. A URI like `/repository/go-proxy/?x=1` is rewritten to `/repository/go-proxy/.+` by this rule instead of falling through to the query-strip rule.

Suggested fix (apply in all four `normalized_uri` blocks):
```yaml
- regexp: "^(/[^/]+/[^/]+/)[^?].*"
  replacement: "$1.+"
```
This still collapses deep paths and consumes any trailing query via `.*`, but leaves `/repository/go-proxy/?x=1` for the second replacement to strip to `/repository/go-proxy/`.

File: tests/e2e/nginx_access_log_exporter_test.go [hunk 2]
```diff
@@ -337,6 +337,47 @@ var _ = Describe("NGINX Access-Log-Exporter Integration", Label("nginx", "monito
 				"Metrics should track different HTTP status codes separately")
 		})
 
+		It("should include normalized_uri label that collapses long paths and preserves short ones", func() {
+			By("Making requests with deep and short URI paths")
+			for _, path := range []string{"/repository/go-proxy/some/deep/path", "/shortpath"} {
```
Comment (yftacherzog) [inline tests/e2e/nginx_access_log_exporter_test.go]: Consider extending coverage to query-string cases, e.g.:
- `/shortpath?foo=bar` → `normalized_uri="/shortpath"`
- `/repository/go-proxy/?x=1` → `normalized_uri="/repository/go-proxy/"` (after the regex fix)

Without these, the query-strip replacement and the Qodo edge case can regress without detection.

**Review by yftacherzog — CHANGES_REQUESTED** [review]
## Review (Yftach's Cursor)

The approach — adding a `normalized_uri` label via access-log-exporter replacements on the tab-separated `$request_uri` field — is sound and fits the cardinality-control goal.

Qodo flagged a correctness issue in the deep-path regex that should be addressed before merge. With first-match-wins replacement semantics, `^(/[^/]+/[^/]+/).+` treats a query-only suffix (e.g. `/repository/go-proxy/?x=1`) as deep-path content and rewrites it to `/repository/go-proxy/.+` instead of stripping the query to `/repository/go-proxy/`. That splits one repository type into two label values based on query presence, which undermines the stated cardinality goal. See inline comment on the regexp.

The e2e test should cover query-string cases, including the edge case above, so this behavior cannot regress silently.

Additional notes:
- Traffic to a 2-segment repo path like `/repository/go-proxy` (no deeper path) will keep that literal label, while deeper paths collapse to `/repository/go-proxy/.+`. Worth confirming this matches dashboard expectations.
- `caching-openshift-e2e-tests / pr group` is failing on this PR; worth checking whether that is related or a flake.

Once the regex and test coverage are updated, the rest looks reasonable.

File: caching/templates/nginx-exporter-configmap.yaml [hunk 3]
```diff
@@ -44,6 +44,16 @@ data:
...
+                    replacement: "$1.+"
+                  - regexp: "\\?.*"
+                    replacement: ""
```
[the remark this answers is not in the document]
Comment (amisstea) [reply caching/templates/nginx-exporter-configmap.yaml:56]: Fixed.

File: tests/e2e/nginx_access_log_exporter_test.go [hunk 4]
```diff
@@ -337,6 +337,47 @@ var _ = Describe("NGINX Access-Log-Exporter Integration", Label("nginx", "monito
...
+				url := testhelpers.GetNginxURL() + path
+				resp, err := nginxClient.Get(url)
+				Expect(err).NotTo(HaveOccurred())
+				resp.Body.Close()
+			}
```
[the remark this answers is not in the document]
Comment (amisstea) [reply tests/e2e/nginx_access_log_exporter_test.go]: Additional test scenarios added.

File: tests/e2e/nginx_access_log_exporter_test.go (hunk 2 above)
Comment (amisstea) [reply tests/e2e/nginx_access_log_exporter_test.go]: Added these cases.

File: caching/templates/nginx-exporter-configmap.yaml (hunk 1 above)
Comment (amisstea) [reply caching/templates/nginx-exporter-configmap.yaml]: Fixed.

**Review by yftacherzog — APPROVED** [review]
